content/components/icon.mdx

```diff
@@ -217,8 +217,40 @@ Maintaining consistency by adhering to the predefined pairs is important to avoi
 
 If you are unable to find the icon you need in our [Octicons](https://primer.style/octicons/) library, you may want to create a new icon. Follow our [Octicons design guidelines](https://primer.style/octicons/guidelines/design) for guidance on creating a new icon that fits with the visual language of our design system. Keep in mind that new icons should be added only when they are necessary and meet the guidelines for use.
 
-## Accessibility
+## Accessibility and usability expectations
+
+If the icon is purely decorative and does not convey any meaning (for instance, if adjacent text already provides all the necessary information/context), it is hidden from screen readers.
+
+If the icon does convey meaning, the component exposes a `role="img"` so that its purpose will be described by assistive technologies, and the component's `aria-label` contains the accessible name / text alternative for the icon.
+
+The accessible name or decision to make it decorative should be included in the design annotation.
+
+Unless the icon is purely decorative, it must have a contrast ratio of at least 3:1 against the background color.
+
+## Built-in accessibility features
+
+If the author does not include an `aria-label`, the component is assumed to be decorative, and will automatically be given an `aria-hidden="true"` attribute.
+
+If the author includes an `aria-label`, the component exposes a `role="img"`, and the `aria-label` attribute is passed to the rendered component.
+
```

Comment:
The information in this section is redundant with the information on line 224. I think we can just remove this section.

Reply:
@alliethu said some redundancy is to be expected